test/lsp.py

```diff
@@ -1,26 +1,57 @@
 #!/usr/bin/env python3
 # pragma pylint: disable=too-many-lines
+# test line 1
 import argparse
 import fnmatch
+import functools
 import json
 import os
+import re
 import subprocess
 import sys
 import traceback
-import re
-import tty
-import functools
 from collections import namedtuple
 from copy import deepcopy
+from enum import Enum, auto
+from itertools import islice
 from pathlib import PurePath
 from typing import Any, List, Optional, Tuple, Union
-from itertools import islice
-
-from enum import Enum, auto
 
-import colorama # Enables the use of SGR & CUP terminal VT sequences on Windows.
+import colorama  # Enables the use of SGR & CUP terminal VT sequences on Windows.
 from deepdiff import DeepDiff
 
+if os.name == 'nt':
+    # pragma pylint: disable=import-error
+    import msvcrt
+else:
+    import tty
+
+
+def escape_string(text: str) -> str:
+    """
+    Trivially escapes given input string's \r \n and \\.
+    """
+    return text.translate(str.maketrans({
+        "\r": r"\r",
+        "\n": r"\n",
+        "\\": r"\\"
+    }))
+
+
+def getCharFromStdin():
+    """
+    Gets a single character from stdin without line-buffering.
+    """
+    if os.name == 'nt':
+        # pragma pylint: disable=import-error
+        return msvcrt.getch().decode("utf-8")
+    else:
+        # Turn off user input buffering so we get the input immediately,
+        # not only after a line break
+        tty.setcbreak(sys.stdin.fileno())
```

Comment:
This doesn't need to be set every time we want a char. Once in the life-time of the program is enough.

Reply:
Exactly, but that way I kept the logic more closely together without over-complifying something simple.